### PR TITLE
c_api and plc binding for lookup src dst using edge ids and type(s)

### PR DESCRIPTION
**Pull request by naimnv**
c_api and plc binding for lookup src dst using edge ids and type(s)

### CODE REVIEW

File: cpp/src/c_api/lookup_src_dst.cpp [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+
+      auto lookup_container = new cugraph::lookup_container_t<edge_t, edge_type_type_t, vertex_t>();
+
+      std::cout << "\n\nGoing to call cugraph::build_edge_id_and_type_to_src_dst_lookup_map\n\n";
+
+      cugraph::build_edge_id_and_type_to_src_dst_lookup_map(
+        handle_, graph_view, edge_ids->view(), edge_types->view());
+
+      std::cout << "\n\nAfter calling cugraph::build_edge_id_and_type_to_src_dst_lookup_map\n\n";
+      /*
```
Comment (ChuckHastings) [inline cpp/src/c_api/lookup_src_dst.cpp:92]: Don't you need this commented out code?

File: cpp/tests/c_api/mg_lookup_src_dst_test.c [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+  TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "create_test_graph failed.");
+  TEST_ALWAYS_ASSERT(ret_code == CUGRAPH_SUCCESS, cugraph_error_message(ret_error));
+
+  ret_code = cugraph_build_edge_id_and_type_to_src_dst_lookup_map(
+    handle, graph, &lookup_container, &ret_error);
+
+  TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, cugraph_error_message(ret_error));
+  TEST_ALWAYS_ASSERT(ret_code == CUGRAPH_SUCCESS, "cugraph_ecg failed.");
+
+#if 0
```
Comment (ChuckHastings) [inline cpp/tests/c_api/mg_lookup_src_dst_test.c:89]: Doesn't the error checking need to happen?

File: cpp/src/c_api/lookup_src_dst.cpp [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+          handle_,
+          *(reinterpret_cast<cugraph::lookup_container_t<edge_t, edge_type_type_t, vertex_t>*>(
+            lookup_container_->lookup_container_)),
+          raft::device_span<edge_t const>(edge_ids_to_lookup_->as_type<edge_t>(),
+                                          edge_ids_to_lookup_->size_),
+          raft::device_span<edge_type_type_t const>(
+            edge_types_to_lookup_->as_type<edge_type_type_t>(), edge_types_to_lookup_->size_));
+
+      auto number_map = reinterpret_cast<rmm::device_uvector<vertex_t>*>(graph_->number_map_);
+
```
Comment (ChuckHastings) [inline cpp/src/c_api/lookup_src_dst.cpp:245]: Don't we need to unrenumber the result src and dst vertices?

File: cpp/src/c_api/lookup_src_dst.cpp [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+        handle_,
+        *(reinterpret_cast<cugraph::lookup_container_t<edge_t, edge_type_type_t, vertex_t>*>(
+          lookup_container_->lookup_container_)),
+        raft::device_span<edge_t const>(edge_ids_to_lookup_->as_type<edge_t>(),
+                                        edge_ids_to_lookup_->size_),
+        edge_type_to_lookup_);
+
+      auto number_map = reinterpret_cast<rmm::device_uvector<vertex_t>*>(graph_->number_map_);
+
+      result_ = new cugraph::c_api::cugraph_lookup_result_t{
```
Comment (ChuckHastings) [inline cpp/src/c_api/lookup_src_dst.cpp:170]: Don't we need to unrenumber the result

**Comment by alexbarghi-nv** [comment]
I'm going to take over this PR since Naim is on leave

File: cpp/src/c_api/lookup_src_dst.cpp [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+extern "C" cugraph_error_code_t cugraph_lookup_endpoints_from_edge_ids_and_single_type(
+  const cugraph_resource_handle_t* handle,
+  cugraph_graph_t* graph,
+  const cugraph_lookup_container_t* lookup_container,
+  const cugraph_type_erased_device_array_view_t* edge_ids_to_lookup,
+  int edge_type_to_lookup,
+  cugraph_lookup_result_t** result,
+  cugraph_error_t** error)
+{
+  lookup_using_edge_ids_of_single_type_functor functor(
```
[the remark this answers is not in the document]
Comment (naimnv) [reply cpp/src/c_api/lookup_src_dst.cpp:362]: Added,

File: cpp/src/c_api/lookup_src_dst.cpp [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+extern "C" cugraph_error_code_t cugraph_lookup_endpoints_from_edge_ids_and_types(
+  const cugraph_resource_handle_t* handle,
+  cugraph_graph_t* graph,
+  const cugraph_lookup_container_t* lookup_container,
+  const cugraph_type_erased_device_array_view_t* edge_ids_to_lookup,
+  const cugraph_type_erased_device_array_view_t* edge_types_to_lookup,
+  cugraph_lookup_result_t** result,
+  cugraph_error_t** error)
+{
+  lookup_using_edge_ids_and_types_functor functor(
```
[the remark this answers is not in the document]
Comment (naimnv) [reply cpp/src/c_api/lookup_src_dst.cpp:328]: Added.

**Comment by alexbarghi-nv** [comment]
Never mind, Naim is able to finish this PR.  Sorry for the confusion.

File: cpp/src/c_api/lookup_src_dst.cpp (hunk 1 above)
Comment (naimnv) [reply cpp/src/c_api/lookup_src_dst.cpp]: Actually all of these were removed, un-renumber added in a commit but never pushed.

**Review by ChuckHastings — APPROVED** [review]

**Review by alexbarghi-nv — APPROVED** [review]
👍

**Comment by alexbarghi-nv** [comment]
Looks like this PR is seeing the OOM error too

**Comment by naimnv** [comment]
> Looks like this PR is seeing the OOM error too

We have been seeing this error occasionally, with different test. I will rerun the failed jobs once the current run finishes.

**Pull request merged**